lib/authorizer/aws4.js

```diff
@@ -78,6 +81,16 @@ var _ = require('lodash'),
                 });
             });
         }
+
+        if (body.mode === RequestBody.MODES.graphql) {
+            graphqlBody = bodyBuilder.graphql(body.graphql).body;
+            hash.update(graphqlBody);
+
+            return callback(hash.digest(digestEncoding));
+        }
+
+        // ensure that callback is called if body.mode doesn't match with any of the above modes
+        return callback();
```

Comment:
with this, we can remove the `body.mode === RequestBody.MODES.formdata` check at the top.